### PR TITLE
Update Pyright config

### PR DESCRIPTION
**Pull request by geordi**

### CODE REVIEW

File: .github/workflows/pyright.yml [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+jobs:
+  pyright:
+    name: pyright
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: jakebailey/pyright-action@v2
+        with:
+          version: 1.1.356
+          github_token: ${{ secrets.GITHUB_TOKEN }} # You need this
```
Comment (Kobzol) [inline .github/workflows/pyright.yml:12]: Where are these options documented? I haven't found it at https://github.com/jakebailey/pyright-action.

Comment (geordi) [reply .github/workflows/pyright.yml:12]: I found it here: https://github.com/jordemort/action-pyright?tab=readme-ov-file#example-usage

But we can experiment with removing `GITHUB_TOKEN`.

Comment (Kobzol) [reply .github/workflows/pyright.yml:12]: Hmm, but you're linking to a different pyright-action, right? The one you link is from `jordemort`, but the GHA workflow uses another one from `jakebailey`.

Comment (geordi) [reply .github/workflows/pyright.yml:12]: Yeah, I combined those two. This PR is basically what we discussed a while ago about trying to make Pyright working in a branch and re-iterating. I'll fix it tomorrow.

**Review by Kobzol — COMMENTED** [review]
CI config looks OK. The error list... less so xD

**Comment by geordi** [comment]
Yeah, I was experimenting a bit with virtual environment config, but it fails on building dependencies. That's something that can be handled in future.

Anyway, the latest Pyright config update seems to be working. Error list is huge...

**Pull request merged**